source/language/scope.rst

```diff
@@ -0,0 +1,61 @@
+Variable scope
+==============
+
+This section describes the rules surrounding scoping of variables in OpenQASM.
+
+Global scope
+------------
+
+Identifiers declared as subroutines, gates, and defcals have global scope, 
+*i.e.* they are visible at every scope of the program, and may not be shadowed.
+Subroutines, gates, and defcals may only be declared at the top-level of the
+program. Redeclaration or overloading of subroutines and gates is not allowed.
+Defcals may be overloaded as described in :any:`pulse-gates`.
+
+Non-quantum variables defined at the top level of the program have global scope,
+and are visible at all lower scopes. Globally-scoped variables are
+implicitly ``const`` inside the body of subroutine and gate definitions, though
+they may be shadowed by locally declared variables or parameters. All qubits
+have global scope, but are not visible within the scope of subroutines, gates,
+and defcals because qubits cannot be interacted with in a manner consistent with
+the ``const`` qualifier.
+
+.. code-block:: c
+
+  int ii = 100; // global scope
+  def myMul() -> int {
+    return ii*2; // ii is in scope, but 'const'
+  }
+  ii = 200;
+  int jj = myMul(); // jj gets the value 400
+
+Block scope
```

Comment:
**question**: How do qubits behave in the context of block scoping? Can we shadow, etc...? In any case, I think we should 
(1) amend the statement on lines 18-21 to reflect that fact 
(2) add an example below.